manila/share/manager.py

```diff
@@ -6010,20 +6002,27 @@ def _server_migration_complete_driver(self, context, source_share_server,
         dest_sn = self.db.share_network_get(context, dest_sn_id)
         dest_sns = self.db.share_network_subnet_get(context, dest_sns_id)
 
+        migration_extended_network_allocations = (
+            CONF.server_migration_extend_neutron_network)
+
         # NOTE(sapcc): Network allocations are extended to the destination host
         # on previous (migration_start) step, i.e. port bindings are created on
         # destination host with existing ports. The network allocations will be
         # cut over on this (migration_complete) step, i.e. port bindings on
         # destination host will be activated and bindings on source host will
-        # be deleted. Since manager takes care of the cutover of network
-        # allocations, driver should not touch them. Therefore the cached
-        # source_share_server and dest_share_server are not updated with the
-        # new network allocations after the cutover_network_allocations call.
-        # Only dest_sns is updated to have the correct segmentation id.
-        if CONF.server_migration_extend_neutron_network:
-            self.driver.network_api.cutover_network_allocations(
-                context, source_share_server, dest_share_server)
-            dest_sns = self.db.share_network_subnet_get(context, dest_sns_id)
+        # be deleted.
+        if migration_extended_network_allocations:
+            allocs = self.driver.network_api.cutover_network_allocations(
+                context, source_share_server)
+            segmentaion_id = self.db.share_server_backend_details_get(
```

Comment:
typo: `segmentation_id`, but consequently also down when it is being used

Reply:
thanks, it's fixed